src/types.jl

```diff
@@ -45,4 +45,10 @@ struct BVPSystem{F <: Function, B <: Union{Function, SciMLBase.TwoPointBVPFuncti
     tmp::tType                  # M
 end
 
-Base.eltype(S::BVPSystem) = eltype(S.tmp)
+Base.eltype(S::BVPSystem) = eltype(S.tmp.du)
+
+# Sparsity Detection
+Base.@kwdef struct MIRKJacobianComputationAlgorithm{BD, CD}
+    bc_diffmode::BD = AutoFiniteDiff()
+    collocation_diffmode::CD = AutoSparseFiniteDiff()
```

Comment:
we normally default to forward diff

Reply:
The forward diff implementation is not entirely efficient yet. I just slapped some `DiffCache`s around to get it to work but it warns that the size of the diffcache is not proper and might hurt performance.

Reply:
resizing will require the diffcache to resize, but that's fine (expected) in this context when the adaptivity occurs. Though what you'd need to add is a `resize!` on the diffcache when the adaptivity happens in order to make it more direct.

Reply:
Yes but here I am getting a resize warning even when the mesh is not resized which makes me think I am somehow incorrectly using DiffCache

Reply:
I think I figured out the problem. I need to use the chunksize of `y`